include/nodes/node.tcc

```diff
@@ -393,6 +400,77 @@ void mpm::Node<Tdim, Tdof, Tnphases>::apply_velocity_constraints() {
   }
 }
 
+//! Assign acceleration constraint
+//! Constrain directions can take values between 0 and Dim * Nphases
+template <unsigned Tdim, unsigned Tdof, unsigned Tnphases>
+bool mpm::Node<Tdim, Tdof, Tnphases>::assign_acceleration_constraint(
+    unsigned dir, double acceleration) {
+  bool status = true;
+  try {
+    //! Constrain directions can take values between 0 and Dim * Nphases
+    if (dir < (Tdim * Tnphases))
+      this->acceleration_constraints_.insert(std::make_pair<unsigned, double>(
+          static_cast<unsigned>(dir), static_cast<double>(acceleration)));
+    else
+      throw std::runtime_error("Constraint direction is out of bounds");
+
```

Comment:
Here, make a check if velocity_constraints in the same direction exist. If yes, throw an error.